lalrpop-util/Cargo.toml

```diff
@@ -11,11 +11,11 @@ version.workspace = true
 rust-version.workspace = true
 
 [dependencies]
-regex = { workspace = true, optional = true }
+regex-automata = { workspace = true, optional = true }
```

Comment:
This would change public feature name `regex` into `regex-automata` but it should be an implementation issue. We can add an alias `[features] regex = ["dep:regex-automata"]` for compatibility and readability, since users just want to enable "regex" use.

Reply:
Nice catch!

Reply:
I double checked that the documentation didn't mention the existence of `regex` feature and enabling it won't make lexer work, so I think it's just an internal detail. I'd like to leave it for now.

Reply:
Public feature renaming is a semvar major change always. And `regex` is mentioned in docs in [4.1 Adding LALRPOP to your project](http://lalrpop.github.io/lalrpop/tutorial/001_adding_lalrpop.html#adding-lalrpop-to-your-cargotoml-file)
> The [dependencies] section describes the dependencies that LALRPOP needs at runtime. All LALRPOP parsers require at least the lalrpop-util crate. In addition, if you don't want to write the lexer by hand, you need to add a dependency on the regex crate.

Reply:
I'm not at a computer to double check, but I'm pretty sure the documentation is out of date on that point. Users adding a regex dependency used to be required to work around some dependency mismanagement, but that's all sorted out now. Users of lalrpop should never need to add a regex dependency unless they actually need it directly.

I'm pretty sure that the use of the regex crate is an internal implement detail to lalrpop, and users should be agnostic of it.